tests/workflow/interfaces/run/test_autograd_run.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for the `run` helper function on the 'autograd' interface"""
+
+from dataclasses import replace
+
+import autograd
+import numpy as np
+import pytest
+
+# pylint: disable=no-name-in-module
+from conftest import atol_for_shots, get_device, test_matrix
+
+import pennylane as qml
+from pennylane import numpy as pnp
+from pennylane.transforms.core import TransformProgram
+from pennylane.workflow import _resolve_execution_config, _setup_transform_program, run
+
+
+@pytest.mark.autograd
+@pytest.mark.parametrize("device, config, shots", test_matrix)
+class TestAutogradRun:
+    """Test the 'autograd' interface run function integrates well for both forward and backward execution"""
+
+    def test_run(self, device, config, shots, seed):
+        """Test execution of tapes on 'autograd' interface."""
+        device = get_device(device, seed=seed)
+        config = replace(config, interface="autograd")
+
+        def cost(a, b):
+            ops1 = [qml.RY(a, wires=0), qml.RX(b, wires=0)]
+            tape1 = qml.tape.QuantumScript(ops1, [qml.expval(qml.PauliZ(0))], shots=shots)
+
+            ops2 = [qml.RY(a, wires="a"), qml.RX(b, wires="a")]
+            tape2 = qml.tape.QuantumScript(ops2, [qml.expval(qml.PauliZ("a"))], shots=shots)
+
+            resolved_config = _resolve_execution_config(
+                config, device, [tape1, tape2], TransformProgram()
+            )
+            inner_tp = _setup_transform_program(TransformProgram(), device, resolved_config)[1]
+            return run([tape1, tape2], device, resolved_config, inner_tp)
+
+        a = pnp.array(0.1, requires_grad=True)
+        b = pnp.array(0.2, requires_grad=False)
+
+        with device.tracker:
+            res = cost(a, b)
+
+        assert len(res) == 2
+
+        if getattr(config, "grad_on_execution", False):
```

Comment:
getattr? Shouldn't always have it?

Reply:
True lol. Consequence of copy-and-paste with sub-par find and replace. 😓